client/valor/client.py

```diff
@@ -224,13 +233,29 @@ def _requests_wrapper(
         url = urljoin(self.host, endpoint)
         requests_method = getattr(requests, method_name)
 
+        timeout_retries = 0
         tried = False
         while True:
             if self.access_token is not None:
                 headers = {"Authorization": f"Bearer {self.access_token}"}
             else:
                 headers = None
-            resp = requests_method(url, headers=headers, *args, **kwargs)
+
+            try:
+                resp = requests_method(
+                    url, headers=headers, timeout=10, *args, **kwargs
+                )
+            except requests.exceptions.Timeout as e:
+                if timeout_retries < max_retries_on_timeout:
+                    time.sleep(
+                        initial_timeout
+                        * exponential_backoff**timeout_retries
+                    )
+                    timeout_retries += 1
+                    continue
+                else:
+                    raise TimeoutError(str(e))
```

Comment:
What is the advantage of this over just `raise e`?

Reply:
it just passes the error message from `Timeout` (a specific exception from `requests.exceptions`) to `TimeoutError` (a generic built-in exception). not necessary, but likely contains additional details beyond the generic exception message

Reply:
Gotcha, makes sense :+1: .